tests/model_serving/model_server/components/kserve_dsc_deployment_mode/utils.py

```diff
@@ -7,63 +7,107 @@
 from simple_logger.logger import get_logger
 from timeout_sampler import TimeoutSampler
 
-from utilities.constants import Timeout
+from kubernetes.dynamic import DynamicClient
+from utilities.constants import Timeout, DscComponents
+from pytest_testconfig import config as py_config
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
 
 LOGGER = get_logger(name=__name__)
 
 
-def wait_for_default_deployment_mode_in_cm(config_map: ConfigMap, deployment_mode: str) -> None:
+def wait_for_configmap_key_value(
+    config_map: ConfigMap,
+    data_key: str,
+    expected_value: Any,
+    nested_key_path: list[str],
+) -> None:
     """
-    Wait for default deployment mode to be set in inferenceservice-config configmap.
+    Wait for a specific value to appear in a configmap under a given key.
 
     Args:
         config_map (ConfigMap): ConfigMap object
-        deployment_mode (str): Deployment mode
-
-    Raises:
-        TimeoutExpiredError: If default deployment mode value is not set in configmap
-
+        data_key (str): Key in configmap.data to parse
+        expected_value (Any): Expected value to wait for
+        nested_key_path (list[str]): Optional list of keys to dig into the nested JSON
     """
-    LOGGER.info(
-        f"Wait for {{request_default_deployment_mode}} deployment mode to be set in {config_map.name} configmap"
-    )
+    LOGGER.info(f"Waiting for {data_key} to be set to {expected_value} in {config_map.name}")
+    nested_key_path = nested_key_path or []
+
     for sample in TimeoutSampler(
         wait_timeout=Timeout.TIMEOUT_5MIN,
         sleep=5,
         func=lambda: config_map.instance.data,
     ):
         if sample:
-            cm_default_deployment_mode = json.loads(sample.deploy)["defaultDeploymentMode"]
-            if cm_default_deployment_mode == deployment_mode:
-                break
+            try:
+                parsed = json.loads(sample.get(data_key, "{}"))
+                for key in nested_key_path:
+                    parsed = parsed.get(key, {})
+                if parsed == expected_value:
+                    LOGGER.info(f"{data_key} set successfully to {expected_value}")
```

Comment:
this log is redundant as the previous log prints what is the check here; if we exit it means it was successful